lab_2_keywords_cooccurrence/main.py

```diff
@@ -84,7 +168,18 @@ def calculate_cumulative_score_for_candidates(candidate_keyword_phrases: KeyPhra
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not(check_sequence(candidate_keyword_phrases, tuple, False) and check_mapping(word_scores, str, float, False)
+           and all(word_scores.get(word) for phrase in candidate_keyword_phrases for word in phrase)):
+        return None
+
+    cmltv_score_dict = {}
+
+    for phrase in set(candidate_keyword_phrases):
+        for word, score in word_scores.items():
```

Comment:
consider iterating over words in a phrase, as phrases are generally shorter than `word_scores`

Reply:
not exactly 

you do not have to iterate over keys of `word_scores`
instead, iterate over words in phrases and ask for the score of that particular word referring to the dictionary
this way, it will be more efficient: you'll perform as many operations as there are words in phrases

the way you do it, you perform <number of words in word_scores> * <number of words in all phrases>

do you agree?